src/promptflow/promptflow/_telemetry/tracer_manager.py

```diff
@@ -0,0 +1,157 @@
+import json
+import logging
+import threading
+from typing import Sequence
+
+from opentelemetry import trace
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import (
+    BatchSpanProcessor,
+    ReadableSpan,
+    SimpleSpanProcessor,
+    SpanExporter,
+    SpanExportResult,
+)
+
+from promptflow._telemetry.mdc_exporter import MdcExporter
+
+logger = logging.getLogger(__name__)
+
+
+class MemoryTraceStore:
```

Comment:
better to make this thread safe, single instance might cause issue in multi-thread scenario.